### ISSUE TITLE
Modular dependencies

### ISSUE THREAD

**Issue by myyk** [seratch/AWScala#68]
Would you consider breaking out the dependencies into more modular pieces? Having to depend on all of aws-java-sdk when only using S3 is really unnecessary. aws-java-sdk is already broken up into pieces, so it would just take breaking the build up into sub-projects and having those depend on each other as needed.

This could potentially increase library adoption. I for one, wouldn't use the library as it is since it's adding something like 85mb of dependencies that I won't be using.

**Comment by seratch** [comment]
>Would you consider breaking out the dependencies into more modular pieces?

Indeed, it's possible but that will make library release heavier than now.

I know some users use this library in serious applications but this project started as a handy tool when working on the Scala REPL. I have other projects to use my time (e.g. Scalatra, json4s, ScalikeJDBC, Skinny Framework). If someone'd like to work on this, that's a quite welcome.

**Comment by howardjohn** [comment]
I took an attempt at this https://github.com/seratch/AWScala/pull/175
